### PR TITLE
fix: fix lint issues (#393)

### PR DESCRIPTION
**Pull request by denys-kolomiitsev**
- Issue #393 

### Related 
Continue to fix in
- Issue: #513

### CODE REVIEW

**Comment by IlyaBondar** [comment]
/deploy-review
>Deployment status: success

File: src/components/Promptbar/Promptbar.tsx [hunk 1]
```diff
@@ -56,6 +56,7 @@ const Promptbar = () => {
   const searchFilters = useAppSelector(PromptsSelectors.selectSearchFilters);
 
   const handleDrop = useCallback(
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (IlyaBondar) [inline src/components/Promptbar/Promptbar.tsx]: Can we add typing here instead of ignoring?

File: src/pages/api/chat.ts [hunk 2]
```diff
@@ -63,6 +63,7 @@ const errorHandler = ({
   msg,
   isStreamingError,
 }: {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (IlyaBondar) [inline src/pages/api/chat.ts]: Can we define error typing instead?

File: src/pages/api/health.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { NextApiRequest, NextApiResponse } from 'next';
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (IlyaBondar) [inline src/pages/api/health.ts]: just use `,res: NextApiResponse)`

File: src/store/addons/addons.reducers.ts [hunk 4]
```diff
@@ -49,6 +49,7 @@ export const addonsSlice = createSlice({
         {} as Record<string, OpenAIEntityAddon>,
       );
     },
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (IlyaBondar) [inline src/store/addons/addons.reducers.ts]: why can't we use `error: { status?: string; statusText?: string }` ?

File: src/store/conversations/conversations.epics.ts [hunk 5]
```diff
@@ -325,6 +325,7 @@ const rateMessageEpic: AppEpic = (action$, state$) =>
       if (!conversation) {
         return of(
           ConversationsActions.rateMessageFail(
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
             (i18n as any).t(
```
Comment (IlyaBondar) [inline src/store/conversations/conversations.epics.ts]: use `translate` instead of `(i18n as any).t`
but problem that method `rateMessageFail` expects `{ error: string | Response; }` which absolutely wrong in this case and should be fixed

File: src/store/conversations/conversations.epics.ts [hunk 6]
```diff
@@ -336,6 +337,7 @@ const rateMessageEpic: AppEpic = (action$, state$) =>
       if (!message || !message.responseId) {
         return of(
           ConversationsActions.rateMessageFail(
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (IlyaBondar) [inline src/store/conversations/conversations.epics.ts]: use `translate` instead of `(i18n as any).t`
but problem that method `rateMessageFail` expects `{ error: string | Response; }` which absolutely wrong in this case and should be fixed

File: src/store/conversations/conversations.epics.ts [hunk 7]
```diff
@@ -725,6 +727,7 @@ const streamMessageEpic: AppEpic = (action$, state$) =>
               ConversationsActions.streamMessageFail({
                 conversation: payload.conversation,
                 message:
+                  // eslint-disable-next-line @typescript-eslint/no-explicit-any
                   (error.cause as any).message ||
```
Comment (IlyaBondar) [inline src/store/conversations/conversations.epics.ts]: `as { message?: string }` ?

File: src/utils/auth/ping-identity.ts [hunk 8]
```diff
@@ -2,6 +2,7 @@ import { OAuthConfig, OAuthUserConfig } from 'next-auth/providers';
 
 // from https://github.com/nextauthjs/next-auth/pull/6614
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
 export interface PingProfile extends Record<string, any> {
```
Comment (IlyaBondar) [inline src/utils/auth/ping-identity.ts]: `Record<string, string | string[] | number>` ?

**Review by IlyaBondar — CHANGES_REQUESTED** [review]
Let's avoid ignoring `@typescript-eslint/no-explicit-any` in as many cases as possible

File: src/utils/server/index.ts [hunk 9]
```diff
@@ -121,6 +121,7 @@ export const OpenAIStream = async ({
   });
 
   if (res.status !== 200) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
     let result: any;
```
Comment (IlyaBondar) [inline src/utils/server/index.ts]: `let result: { error?: OpenAIError }`

Comment (denys-kolomiitsev) [reply src/utils/server/index.ts]: Changed.
Thank you for the suggestion.

File: src/utils/auth/ping-identity.ts (hunk 8 above)
Comment (denys-kolomiitsev) [reply src/utils/auth/ping-identity.ts]: Changed.
Thank you for the suggestion.

File: src/store/conversations/conversations.epics.ts (hunk 7 above)
Comment (denys-kolomiitsev) [reply src/store/conversations/conversations.epics.ts]: Changed to the different type.

File: src/store/conversations/conversations.epics.ts (hunk 6 above)
Comment (denys-kolomiitsev) [reply src/store/conversations/conversations.epics.ts]: Fixed as advised

File: src/store/conversations/conversations.epics.ts (hunk 5 above)
Comment (denys-kolomiitsev) [reply src/store/conversations/conversations.epics.ts]: Fixed as advised

File: src/store/addons/addons.reducers.ts (hunk 4 above)
Comment (denys-kolomiitsev) [reply src/store/addons/addons.reducers.ts]: Fixed.

File: src/pages/api/health.ts (hunk 3 above)
Comment (denys-kolomiitsev) [reply src/pages/api/health.ts]: changed to the NextApiResponse

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Deployment status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review

>Deployment status: success

File: src/components/Search/SearchFilterRenderer.tsx [hunk 10]
```diff
@@ -11,7 +11,7 @@ export default function SearchFilterRenderer({
   dataQa,
   ...props
 }: CustomTriggerMenuRendererProps) {
-  const [checked, setChecked] = useState(isSelected);
+  const [checked, setChecked] = useState<boolean>(isSelected as boolean);
```
Comment (IlyaBondar) [inline src/components/Search/SearchFilterRenderer.tsx]:
```suggestion
  const [checked, setChecked] = useState(!!isSelected);
```

File: src/store/models/models.reducers.ts [hunk 11]
```diff
@@ -51,6 +51,7 @@ export const modelsSlice = createSlice({
         {} as Record<string, OpenAIEntityModel>,
       );
     },
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
     getModelsFail: (state, { payload }: PayloadAction<{ error: any }>) => {
```
Comment (IlyaBondar) [inline src/store/models/models.reducers.ts]:
```suggestion
    getModelsFail: (state, { payload }: PayloadAction<{ error: { status?: string; statusText?: string }}>) => {
```

File: src/utils/server/index.ts [hunk 12]
```diff
@@ -196,6 +200,7 @@ export const OpenAIStream = async ({
 
       const parser = createParser(onParse);
 
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
       for await (const chunk of res.body as any) {
```
Comment (IlyaBondar) [inline src/utils/server/index.ts]: if(res.body) {
        for await (const chunk of res.body) {
          if (isFinished) {
            return;
          }
          parser.feed(decoder.decode(chunk as Buffer));
        }
      }

File: src/utils/auth/nextauth.ts [hunk 13]
```diff
@@ -107,6 +108,7 @@ async function refreshAccessToken(token: JWT & any) {
       token: returnToken,
     });
     return returnToken;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
   } catch (error: any) {
```
Comment (IlyaBondar) [inline src/utils/auth/nextauth.ts]:
```suggestion
  } catch (error: { message?:string }) {
```

File: src/utils/auth/nextauth.ts [hunk 14]
```diff
@@ -34,6 +34,7 @@ export const tokenConfig: TokenEndpointHandler = {
  * `accessToken` and `accessTokenExpires`. If an error occurs,
  * returns the old token and an error property
  */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
 async function refreshAccessToken(token: JWT & any) {
```
Comment (IlyaBondar) [inline src/utils/auth/nextauth.ts]:
```suggestion
async function refreshAccessToken(token: JWT & { providerId?: string, userId:string }) {
```

**Review by IlyaBondar — CHANGES_REQUESTED** [review]
@denys-kolomiitsev , there are 4 warnings
image
and 15 places where "@typescript-eslint/no-explicit-any" was ignored - need to try to add typing where it's possible:
image

File: src/components/Promptbar/Promptbar.tsx [hunk 15]
```diff
@@ -56,15 +56,15 @@ const Promptbar = () => {
...
+    (e: DragEvent<HTMLDivElement>) => {
       if (e.dataTransfer) {
         const prompt = JSON.parse(e.dataTransfer.getData('prompt'));
 
         dispatch(
           PromptsActions.updatePrompt({
             promptId: prompt.id,
             values: {
-              folderId: e.target.dataset.folderId,
+              folderId: e.currentTarget.dataset.folderId,
```
Comment (mikitabut) [inline src/components/Promptbar/Promptbar.tsx:67]: Just to be sure sure - is this change needed?

File: src/pages/api/files/file/[...slug].ts [hunk 16]
```diff
@@ -76,10 +76,10 @@ async function handlePutRequest(
     body: readable,
   });
 
-  const json: any = await proxyRes.json();
+  const json: unknown = await proxyRes.json();
   if (!proxyRes.ok) {
     throw new OpenAIError(
-      json || proxyRes.statusText,
+      (typeof json === 'string' && json) || proxyRes.statusText,
```
Comment (mikitabut) [inline src/pages/api/files/file/[...slug].ts:82]: you've changed the logic here - previously json object passed inside, but now only string
Please recheck and revert it back

File: src/pages/api/files/file/[...slug].ts [hunk 17]
```diff
@@ -144,9 +144,9 @@ async function handleDeleteRequest(
   });
 
   if (!proxyRes.ok) {
-    const json: any = await proxyRes.json();
+    const json: unknown = await proxyRes.json();
     throw new OpenAIError(
-      json || proxyRes.statusText,
+      (typeof json === 'string' && json) || proxyRes.statusText,
```
Comment (mikitabut) [inline src/pages/api/files/file/[...slug].ts:149]: same

File: src/store/addons/addons.reducers.ts [hunk 18]
```diff
@@ -49,11 +49,20 @@ export const addonsSlice = createSlice({
...
+      }: PayloadAction<{
+        error: { status?: string | number; statusText?: string };
+      }>,
+    ) => {
       state.isLoading = false;
       state.error = {
         title: translate('Error fetching addons.'),
-        code: payload.error.status || 'unknown',
+        code:
+          (payload.error.status && payload.error.status.toString()) ||
```
Comment (mikitabut) [inline src/store/addons/addons.reducers.ts]: nit: looks like you can use just `payload.error.status?.toString() ?? 'unknown'`

File: src/store/models/models.reducers.ts [hunk 19]
```diff
@@ -51,11 +51,20 @@ export const modelsSlice = createSlice({
...
+      }: PayloadAction<{
+        error: { status?: string | number; statusText?: string };
+      }>,
+    ) => {
       state.isLoading = false;
       state.error = {
         title: translate('Error fetching models.'),
-        code: payload.error.status || 'unknown',
+        code:
+          (payload.error.status && payload.error.status.toString()) ||
```
Comment (mikitabut) [inline src/store/models/models.reducers.ts]: same here

File: src/utils/app/__tests__/forms.test.tsx [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+            await userEvent.click(submitButton);
+
+            const validationMessageElement = queryByText(formValid);
+
+            expect(validationMessageElement).not.toBeInTheDocument();
+
+
+        });
+    })
+})
```
Comment (mikitabut) [inline src/utils/app/__tests__/forms.test.tsx]: You don't have empty line at the end of file
Could you please recheck eslint rules and add the rule for empty line at the end?

File: src/utils/app/__tests__/forms.test.tsx [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+            const submitButton = getByText(buttonText);
+
+            await userEvent.type(input, 'value 1');
+
+            await userEvent.click(submitButton);
+
+            const validationMessageElement = queryByText(formValid);
+
+            expect(validationMessageElement).not.toBeInTheDocument();
+
```
Comment (mikitabut) [inline src/utils/app/__tests__/forms.test.tsx:119]: Too much empty lines one by one
Please check eslint rules and fix it if possible

File: src/utils/app/__tests__/forms.test.tsx [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+                    <input
+                        checked={isChecked}
+                        onChange={(e) => { setIsChecked(e.currentTarget.checked) }}
+                        id='checkbox-id'
+                        type='checkbox'
+                        name={checkboxName}
+                        ref={checkboxRef}
+                        required />
+                </label>
+
```
Comment (mikitabut) [inline src/utils/app/__tests__/forms.test.tsx]: Too much empty lines one by one
Please check eslint rules and fix it if possible

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Deployment status: success

File: src/components/Common/FolderContextMenu.tsx [hunk 23]
```diff
@@ -73,21 +73,21 @@ export const FolderContextMenu = ({
         display: !!onUpload && !isExternal,
         dataQa: 'upload',
         Icon: IconUpload,
-        onClick: onUpload,
+        onClick: onUpload as onClickMenuItemHandler,
```
Comment (mikitabut) [inline src/components/Common/FolderContextMenu.tsx:76]: Is this assertion really needed? Maybe we need to have type which will not need type assertions? Create TODO: here in such case and create techdebt ticket to fix it in future

**Review by mikitabut — APPROVED** [review]

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Deployment status: success

**Review by IlyaBondar — APPROVED** [review]
